### PR TITLE
Add MirrorHelper script to sync upstream Github repos to downstream servers

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

File: config/mirror.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+:branch_mirror_defaults: {}
+:branch_mirror_overrides: {}
+:productization_name: manageiq
+:remotes:
+  :red_hat_cloudforms: git@github.com:RedHatCloudForms
+  :upstream: git@github.com:ManageIQ
+:repos_to_mirror:
+  cfme-cloud_services: :red_hat_cloudforms
+  cfme-migration_analytics: :red_hat_cloudforms
```
Comment (Fryguy) [inline config/mirror.yml]: We should drop these, then leverage config gem merging of yaml to allow extensions via something like mirror.custom.yml

Comment (bdunne) [reply config/mirror.yml]: Ok, I wasn't sure if anything depended on these two

Comment (bdunne) [reply config/mirror.yml]: Done

File: lib/manageiq/release/mirror_helper.rb [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+          sync_branch(source_remote, source_name, dest_remote, dest_name)
+        end
+      end
+
+      def mirror_upstream_repo(repo)
+        mirror_remote_refs("upstream", "downstream")
+        mirror_branches(repo, "upstream", "downstream")
+        mirror_remote_refs("downstream", "backup")
+      end
+      alias_method :mirror_red_hat_cloudforms_repo, :mirror_upstream_repo
```
Comment (Fryguy) [inline lib/manageiq/release/mirror_helper.rb]: Do we need this?

Comment (bdunne) [reply lib/manageiq/release/mirror_helper.rb]: Removed...

**Comment by bdunne** [comment]
@Fryguy Any other concerns?

**Comment by chessbyte** [comment]
@Fryguy bump

File: lib/manageiq/release/mirror_helper.rb [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  def green;  "\e[32m#{self}\e[0m" end
+  def yellow; "\e[33m#{self}\e[0m" end
+  def cyan;   "\e[36m#{self}\e[0m" end
+  def bold;   "\e[1m#{self}\e[22m" end
+end
+
+require 'config'
+Config.load_and_set_settings(
+  Dir.glob(ManageIQ::Release::CONFIG_DIR.join("mirror*.yml")).sort.reverse
+)
```
Comment (Fryguy) [inline lib/manageiq/release/mirror_helper.rb]: Not a fan of these things happening at the `lib/manageiq/release/mirror_helper.rb` level as just requiring this file ends up setting it globally (i.e. even in unrelated stuff).  IMO mirror_helper.rb should just have the module and then some bin script does the config setup (or it's passed in as a parameter and isolated to the mirrorheper instead of globally changing things)

File: lib/manageiq/release/mirror_helper.rb [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        end
+
+        puts
+      end
+
+      def clone_repo(upstream_repo, downstream_repo, path, remote_source)
+        system("git clone #{Settings.remotes[remote_source]}/#{upstream_repo}.git #{path} -o upstream")
+        Dir.chdir(path) do
+          system("git remote add downstream #{Settings.remotes.downstream}/#{downstream_repo}.git") unless remote_exists?("downstream")
+          system("git remote add backup #{Settings.remotes.backup}/#{downstream_repo}.git")
```
Comment (Fryguy) [inline lib/manageiq/release/mirror_helper.rb:98]: If you don't have a backup, then this adds a weird remote.  This should probably be protected with an `if Settings.remotes.backup`

Comment (Fryguy) [reply lib/manageiq/release/mirror_helper.rb:98]: or, since there's a double-check on downstream, we probably needs something here as well, so...

```ruby
if Settings.remotes.backup && !remote_exists?("backup")
```

Comment (bdunne) [reply lib/manageiq/release/mirror_helper.rb]: Done

File: lib/manageiq/release/mirror_helper.rb (hunk 3 above)
Comment (bdunne) [reply lib/manageiq/release/mirror_helper.rb]: I switched to a class, so now they are done on initialize.  This changes the interface a little bit, but I think it's for the best.
